File: Java_ChildrenCompetition/src/main/java/repository/ITestParticipantRelationRepository.java
package repository;

import model.TestParticipantRelation;
import model.Tuple;

//public interface ITestParticipantRelationRepository<ID, T extends HasId<ID>> extends IRepository<ID, T> {
//
//}

public interface ITestParticipantRelationRepository extends IRepository<Tuple<Integer, Integer>, TestParticipantRelation> {

}

File: CSharp_ChildrenCompetitionSockets/CSharp_ChildrenCompetitionSockets/persistence/ITestParticipantRelationRepository.cs
using System;
using System.Collections.Generic;
using CSharp_ChildrenCompetitionGUI.model;

namespace CSharp_ChildrenCompetitionGUI.repository
{
    public interface ITestParticipantRelationRepository : IRepository<Tuple<int, int>, TestParticipantRelation>
    {

    }
}